### ISSUE TITLE
Add new touchpad gestures

### ISSUE THREAD

**Issue by bala7s** [pop-os/cosmic-epoch#1367]
Currently only 4 finger left/right has a function, to swap between workspaces. 4 finger up could open the workspaces view and 4 finger down could close it. In macOS I really love touchpad gestures making me use the touchpad even when I use separate mouse too. I would like to see this in Cosmic too.

**Comment by git-f0x** [comment]
It's planned, but isn't yet implemented. The other 4 finger directions will be used for workspaces and the App Library, and there'll also be 3 finger gestures (to switch app focus).

**Comment by ThomasCorbin** [comment]
Will these be configurable?

**Comment by leviport** [comment]
Of course

**Comment by raphmim** [comment]
Very excited for these :) Hoping it makes it into the Beta!

**Comment by tristonarmstrong** [comment]
> Very excited for these :) Hoping it makes it into the Beta!

oof.. bad news

**Comment by usbeehu** [comment]
Hopefully we will see this in Epoch 2.

**Comment by wompeter** [comment]
definitely would like to see configurable gestures. I prefer a 3-finger swipe left and right for workspaces. I'm trying to get used to 4-fingers right now and it's making my brain uncomfortable

**Comment by tristonarmstrong** [comment]
> definitely would like to see configurable gestures. I prefer a 3-finger swipe left and right for workspaces. I'm trying to get used to 4-fingers right now and it's making my brain uncomfortable

agreed, four finger swipe is just not how its done. Everyone else does a 3 finger swipe for a reason and im willing to bet it has some to due with consistency and accessibility.

**Comment by usbeehu** [comment]
4 finger is good for certain things (see MacOS) but I would prefer 3 finger gestures too!

**Comment by tristonarmstrong** [comment]
> 4 finger is good for certain things (see MacOS) but I would prefer 3 finger gestures too!

Just checked with my macbook.. Had no idea there were 4 finger gestures, tho they seem redundant.

**Comment by usbeehu** [comment]
I use [them](https://support.apple.com/en-us/102482) often, pretty handy.

<img width="841" height="454" alt="Image" src="https://github.com/user-attachments/assets/f676385c-3d0b-4f97-950e-68dd5cffae5c" />

**Comment by tristonarmstrong** [comment]
Yeah i looked, these are the defaults, but the four finger swipe does exist for sure as alternatives.

<img width="751" height="811" alt="Image" src="https://github.com/user-attachments/assets/ac8cc7e4-f54c-4762-8191-47ab2053b33f" />

**Comment by hs-ye** [comment]
I was able to get touchegg https://github.com/joseexposito/touchegg working after installing it and having my code agent look at it, and then set my own configs for it.

It was a bit of a faff as it conflicts with the built-in popos fork of touchegg (my agent decided to turn off the pop-os versoin), but installing the stock touchegg gives you control over the config of how to define what each action does.
